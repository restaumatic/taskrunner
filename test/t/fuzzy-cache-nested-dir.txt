# no toplevel
# s3

# Check how caching works when we're not running directly in root dir.
# We should still be able to cache stuff outside cwd (but inside root dir).

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir -p a

(
  cd a
  git init -q
  mkdir nested
  cd nested
  echo foo > input.txt
  git add input.txt
)

cp -r a b

go() {
  # Note: we want separate workdir for separate repos
  export TASKRUNNER_STATE_DIRECTORY="$(pwd)" TASKRUNNER_ROOT_DIRECTORY="$(pwd)"
  (
    cd nested
    taskrunner -n mytask bash -e -c '
      snapshot input.txt --fuzzy-cache --outputs output.txt ../output2.txt
      echo -n "Expensive computation, "
      if [ -f output.txt ]; then
        echo "previous output is $(cat output.txt)"
      else
        echo "no previous output"
      fi
      rev input.txt > output.txt
      cp output.txt ../output2.txt
    '
  )
}

(
  cd a
  go
  echo "Output in a is: $(cat nested/output.txt)"
  echo "Output2 in a is: $(cat output2.txt)"
)

(
  cd b
  echo bar > nested/input.txt
  go ../out_b # note: should run "expensive computation" again, but having previous output.txt available
  echo "Output in b is: $(cat nested/output.txt)"
  echo "Output2 in b is: $(cat output2.txt)"
)
