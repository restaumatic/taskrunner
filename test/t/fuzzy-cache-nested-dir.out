-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation, no previous output
[mytask] info | success
Output in a is: oof
Output2 in a is: oof
[mytask] info | Inputs changed, running task
[mytask] info | Restored fuzzy cache from branch master, hash=c6621d9fa9e14eff5f583847f3d3463767c6c082
[mytask] stdout | Expensive computation, previous output is oof
[mytask] info | success
Output in b is: rab
Output2 in b is: rab
